plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * Helper function to detect if static assets have far-future expiration headers.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Callback for the far-future caching test.
+ *
+ * @since n.e.x.t
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function perflab_ffh_assets_test(): array {
+	$result = array(
+		'label'       => __( 'Your site serves static assets with far-future expiration headers', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__(
+				'Serving static assets with far-future expiration headers improves performance by allowing browsers to cache files for a long time, reducing repeated requests.',
+				'performance-lab'
+			)
+		),
+		'actions'     => '',
+		'test'        => 'is_far_future_headers_enabled',
+	);
+
+	// List of assets to check.
+	$assets = array(
+		includes_url( 'js/wp-embed.min.js' ),
+		includes_url( 'css/buttons.min.css' ),
+		includes_url( 'fonts/dashicons.woff2' ),
+		includes_url( 'images/media/video.png' ),
+	);
+
+	/**
+	 * Filters the list of assets to check for far-future headers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string[] $assets List of asset URLs to check.
+	 */
+	$assets = apply_filters( 'perflab_ffh_assets_to_check', $assets );
```

Comment:
Since a filter could make it return anything.

```suggestion
	$assets = apply_filters( 'perflab_ffh_assets_to_check', $assets );
	$assets = array_filter( (array) $assets, 'is_string' );
```